internal/docstore/doc.go

```diff
@@ -13,38 +13,142 @@
+// - Create creates a new document.
+// - Replace replaces an existing document.
+// - Put puts a document into a collection whether or not it is already present.
+// - Update applies a set of modifications to a document.
+// - Delete deletes a document.
+//
+// Each action acts atomically on a single document. You can execute actions
+// individually--the Collection type has a method for each one--or you can group them
+// into a list, like so:
+//
+//   err := coll.Actions().Put(doc1).Replace(doc2).Get(doc3).Do(ctx)
+//
+// When you use an action list, docstore will try to optimize the execution of the
+// actions. For example, multiple Get actions may be combined into a single "batch
+// get" RPC. For the most part, actions in a list execute in an undefined order
+// (perhaps concurrently) and independently, but a Get and a write on the same
+// document are ordered with respect to each other. See the documentation of
+// ActionList for details.
 //
 //
 // Revisions
 //
-// Docstore gives every document a revision when it is created. Docstore uses the
-// field name "DocstoreRevision" (stored in the constant docstore.RevisionField) to
-// hold the revision. Whenever a document is modified, its revision changes.
-// Revisions can be used for optimistic locking: whenever a Put, Replace, Update or
-// Delete action is given a document with a revision, then an error for which
-// gcerrors.Code returns FailedPrecondition or NotFound is
-// returned if the stored document's revision does not match the given document's.
-// Thus a Get followed by one of those write actions will fail if the document was
-// changed between the Get and the write.
+// Docstore gives every document a revision when it is created. Whenever a document
+// is modified, its revision changes. Revisions can be used for optimistic locking:
+// whenever you pass a document with a revision to Put, Replace, Update or Delete,
+// Docstore will return an error with code FailedPrecondition if the stored
+// document's revision does not match the given document's. (See the
+// gocloud.dev/gcerrors package for information about error codes.) A Get followed by
+// one of those write actions will fail if the document was changed between the Get
+// and the write.
+//
+// By default, Docstore uses the field name "DocstoreRevision" (stored in the
+// constant docstore.RevisionField) to hold the revision. Some providers give you the
+// option of changing that field name.
+// TODO(jba): make it all providers
 //
 // Since different providers use different types for revisions, the type of the
 // revision field is unspecified. When defining a struct for storing docstore data,
 // define the field to be of type interface{}. For example,
-//    type User { Name string; DocstoreRevision interface{} }
+//
+//    type User {
+//      Name             string
+//      DocstoreRevision interface{}
+//    }
+//
 // If a struct doesn't have a DocstoreRevision field, then the logic described above
```

Comment:
I feel like this caveat should be moved up to the beginning of the section. Something more like (please wordsmith, just suggesting the order):

Docstore supports revisions of documents based on a document field. By default it uses the field "DocstoreRevision" to hold the revision. Some providers give you the option....

If a struct doesn't have a revision field, no revisioning is performed. Otherwise, every document is given a revision when created .... etc.

Reply:
Done.